ui/balances_table.ts

```diff
@@ -0,0 +1,261 @@
+                usd_value: value
+            };
+        }
+    }
+
+    // now let's get the total usd
+    let total_usd = 0;
+    for (const asset in inter_data) {
+        if (!inter_data.hasOwnProperty(asset)) {
+            continue;
+        }
+        total_usd += inter_data[asset]['usd_value'] as number;
+    }
+
+    const full_data: AssetBalance[] = [];
+    // we should have all balances by now
+    for (const asset in inter_data) {
+        if (!inter_data.hasOwnProperty(asset)) {
+            continue;
+        }
+        const amount = inter_data[asset].amount as number;
+        const amountStr = amount.toFixed(settings.floating_precision);
+        const value = inter_data[asset].usd_value as number;
+        const percentage = value / total_usd;
+        const percentageStr = (percentage * 100).toFixed(settings.floating_precision);
+        const valueStr = value.toFixed(settings.floating_precision);
+        full_data.push({
+            asset: asset,
+            amount: amountStr,
+            usd_value: valueStr,
+            percentage: percentageStr
+        });
+    }
+    return full_data;
+}
+
+export function total_balances_get() {
+    return SAVED_BALANCES;
+}
+
+export function total_table_recreate() {
+    // only create the total table if we are in the dashboard
+    if (settings.current_location !== 'index') {
+        return;
+    }
+
+    const full_data = create_full_data();
+
+    if (!TOTAL_BALANCES_TABLE) {
+        init_balances_table(full_data);
+    } else {
+        if (!$('#table_balances_total_body').length) {
+            // if table has already been initialized but no longer existing on the page
+            TOTAL_BALANCES_TABLE.destroy(false);
+            init_balances_table(full_data);
+        } else {
+            // update the already existing table
+            TOTAL_BALANCES_TABLE.clear();
+            TOTAL_BALANCES_TABLE.rows.add(full_data);
+            balance_table_init_callback.call(TOTAL_BALANCES_TABLE);
+            $(TOTAL_BALANCES_TABLE.column(2).header()).text(`${settings.main_currency.ticker_symbol} value`);
+            TOTAL_BALANCES_TABLE.draw();
+        }
+    }
+    update_last_balance_save();
+}
+
+export function total_table_add_balances(location: string, query_result: { [asset: string]: AssetBalance }) {
+    const data: { [asset: string]: AssetBalance } = {};
+    for (const asset in query_result) {
+        if (!query_result.hasOwnProperty(asset)) {
+            continue;
+        }
+
+        if (asset === 'location' || asset === 'net_usd') {
+            continue;
+        }
+
+        const amount = parseFloat(query_result[asset].amount.toString());
+        const value = parseFloat(query_result[asset].usd_value.toString());
+        data[asset] = {amount: amount, usd_value: value};
+    }
+    SAVED_BALANCES[location] = data;
+    total_table_recreate();
+}
+
+function balance_table_init_callback(sett?: DataTables.Api, _json?: any) {
```

Comment:
why `sett` and not `settings` here? `sett` is a terrible name. Why did you rename it?

Reply:
I had a warning due to settings being already a globally declared constant. 
I will change it to something else :)